### DISCUSSION TITLE
Is there any way to draw some waveform diagrams

### DISCUSSION THREAD

**Discussion by Crazy-Creeper** [fyne-io/fyne#3740] [Q&A]
I have two ideas, the first one is to draw a chart based on the coordinate points filled in, and the second one is to draw a chart by dragging the mouse. I'm wondering if there is a way to render irregular shapes, and if the second approach is feasible.

**Comment by andydotxyz** [comment]
We don't have builtin API for drawing arbitrary vector paths, but you can do it with pixels and a `Raster` canvas object.
I'd recommend checking out https://addons.fyne.io/addons/berlin.martyn.waveform.html
